Segfault when using undeclared identifier

Compiling the following (incorrect) C code that is using undeclared identifier with bcc (commit dfb0d98c8f03a4701839503ca29c7d4986196b88) results in segfault

```c
#include <uapi/linux/ptrace.h>
#include <linux/sched.h>

struct data_t {
    int pid;
};

/* BPF_RINGBUF_OUTPUT(output, ...) declaration is missing */
int count_sched(struct pt_regs *ctx) {
    struct data_t *data = output.ringbuf_reserve(sizeof(struct data_t));
    output.ringbuf_discard(data, 0);
    return 0;
}
```

The stdout/stderr output looks as follow

```
/virtual/main.c:9:27: error: use of undeclared identifier 'output'
    struct data_t *data = output.ringbuf_reserve(sizeof(struct data_t));
                          ^
/virtual/main.c:10:5: error: use of undeclared identifier 'output'
    output.ringbuf_discard(data, 0);
    ^
Segmentation fault (core dumped)
```

And from the stack-trace it looks like it comes down to `CallExpr.getCalleeDecl()` returning NULL in this case.

--- a/src/cc/frontends/clang/b_frontend_action.cc
+++ b/src/cc/frontends/clang/b_frontend_action.cc
@@ -209,7 +209,7 @@ class ProbeChecker : public RecursiveASTVisitor<ProbeChecker> {
 
     if (!track_helpers_)
       return false;
-    if (VarDecl *V = dyn_cast<VarDecl>(E->getCalleeDecl()))
+    if (VarDecl *V = dyn_cast_or_null<VarDecl>(E->getCalleeDecl()))
       needs_probe_ = V->getName() == "bpf_get_current_task";
     return false;
   }
@@ -421,8 +421,12 @@ bool ProbeVisitor::TraverseStmt(Stmt *S) {
 }
 
 bool ProbeVisitor::VisitCallExpr(CallExpr *Call) {
+  Decl *decl = Call->getCalleeDecl();
+  if (decl == nullptr)
+      return true;
+
   // Skip bpf_probe_read for the third argument if it is an AddrOf.
-  if (VarDecl *V = dyn_cast<VarDecl>(Call->getCalleeDecl())) {
+  if (VarDecl *V = dyn_cast<VarDecl>(decl)) {
     if (V->getName() == "bpf_probe_read" && Call->getNumArgs() >= 3) {
       const Expr *E = Call->getArg(2)->IgnoreParenCasts();
       whitelist_.insert(E);
@@ -430,7 +434,7 @@ bool ProbeVisitor::VisitCallExpr(CallExpr *Call) {
     }
   }
 
-  if (FunctionDecl *F = dyn_cast<FunctionDecl>(Call->getCalleeDecl())) {
+  if (FunctionDecl *F = dyn_cast<FunctionDecl>(decl)) {
     if (F->hasBody()) {
       unsigned i = 0;
       for (auto arg : Call->arguments()) {

